CRM/Multicompanyaccounting/Hook/ValidateForm/FinancialTypeAccount.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+/**
+ * Form Validation on editing or assign a financial account for a type
+ */
+class CRM_Multicompanyaccounting_Hook_ValidateForm_FinancialTypeAccount {
+
+  private $form;
+  private $fields;
+  private $errors;
+
+  public function __construct(&$form, &$fields, &$errors) {
+    $this->form = $form;
+    $this->fields = &$fields;
+    $this->errors = &$errors;
+  }
+
+  /**
+   * Validates the financial type account form
+   * when editing or assigning a new account
+   */
+  public function validate() {
+    $this->validateFinancialAccountOrangistion();
+  }
+
+  /**
+   * Gets the owner of the financial account from the submitted fields
+   */
+  private function getOwnerContactIdFromFinancialAccountField() {
```

Comment:
better for readability to have `getOwnerContactIdFromFinancialAccountField` and `getFinancialAccountByFinancialTypeId` defined after `validateFinancialAccountOrangistion`,  in the order they are called in.